automations/lifter.py

```diff
@@ -8,25 +8,43 @@ class LifterAutomation(StateMachine):
     lifter: Lifter
     intake: Intake
 
-    @state(first=True)
-    def move(self):
-        """Move to lifter height according to button press(5 buttons)"""
-        self.lifter.move()
-        self.next_state("move_complete")
-
-    @state()
-    def move_complete(self):
-        """This state makes sure that you are at you set height."""
-        if self.lifter.at_pos():
-            self.next_state("eject")
+    @state(must_finish=True)
+    def move_upper_scale(self):
+        self.lifter.default_height = self.lifter.UPPER_SCALE
+        self.next_state("move")
+
+    @state(must_finish=True)
+    def move_balanced_scale(self):
+        self.lifter.default_height = self.lifter.BALANCED_SCALE
+        self.next_state("move")
+
+    @state(must_finish=True)
+    def move_lower_scale(self):
+        self.lifter.default_height = self.lifter.LOWER_SCALE
+        self.next_state("move")
+
+    @state(must_finish=True)
+    def move_switch(self):
+        self.lifter.default_height = self.lifter.SWITCH_HEIGHT
+        self.next_state("move")
+
+    @state(first=True, must_finish=True)
+    def move(self, initial_call):
+        """Move to lifter height according to default height"""
+        if initial_call:
+            self.lifter.move()
+
+        if self.lifter.at_pos() and self.lifter.at_pos_switch_pressed():
```

Comment:
The way you are doing this is problematic - if the profile you are executing ends short of the limit switch, the motor will be stopped, but this condition will not be true as you are not tripping the limit switch. Need a new approach.